src/HotChocolate/Core/src/Types/Types/Descriptors/Conventions/XmlDocumentationProvider.cs

```diff
@@ -31,28 +33,75 @@ public class XmlDocumentationProvider
             {
                 throw new ArgumentNullException(nameof(fileResolver));
             }
+
             _fileResolver = fileResolver;
         }
 
-        public string GetSummary(Type type) =>
-            GetSummary((MemberInfo)type);
+        public string GetDescription(Type type) =>
+            GetDescription((MemberInfo)type);
 
-        public string GetSummary(MemberInfo member)
+        public string GetDescription(MemberInfo member)
         {
             var assemblyName = member.Module.Assembly.GetName();
             var element = GetMemberElement(member);
-            return RemoveLineBreakWhiteSpaces(
-                GetText(element?.Element(_summaryElementName)));
+
+            XElement summary = element?.Element(_summaryElementName);
+
+            XElement returns = element?.Element(_returnsElementName);
+
+            var exceptions = element?.Elements(_exceptionElementName).ToList();
```

Comment:
Do we really need this list instance? In large schemas with 30000+ this will create a lot of instances that we do not need since we just iterate over it in a local function.

Reply:
what would you suggest to use here? I had to convert it to a list to be able to check if it has any values. At least Rider warned about performance implications if I would use the "Count" or "Any" methods to check if the IEnumerable was empty. As far as I remember the exceptions var wouldn't be null even it there were no exception elements.

Reply:
I rewrite it to be more efficient ... we will optimize it further with the BlockString-Algorithm in a separate PR. Have a look at the new code to see what we optimized.